pkg/upgrade/upgrademanager/manager.go

```diff
@@ -346,6 +368,10 @@ func (m *Manager) Migrate(
 		return nil
 	}
 
+	// We only need to write the first fence to the settings table for
+	// secondary tenant upgrade.
+	writtenFirstFence := m.codec.ForSystemTenant()
```

Comment:
nit: do I understand correctly that the system tenant doesn't actually write the first fence, it's just that it doesn't need to? If so, what do you think of calling this variable, `mustWriteFenceVersion` and default it to false for the system tenant.